### ISSUE TITLE
Command-line argument -n/--no-compile for lfc not working for federated programs

### ISSUE THREAD

**Issue by siljesu** [lf-lang/lingua-franca#1598]
On the master branch, reproduce the error by running `lfc -n` or `lfc --no-compile` with any federated program. It runs as if the option is ignored.

Specifying `no-compile: true` as a target property works as expected on the master branch.

Tested with `lf-cli-nightly-20230222-050500` and the following simple federated program:

```
target C {
    timeout: 2 secs,
}

reactor Fed {
    input in: int
    output out: int
}

federated reactor {
    fed1 = new Fed()
    fed2 = new Fed()

    fed1.out -> fed2.in
    fed2.out -> fed1.in
}
```

**Comment by lhstrh** [comment]
Thanks for the bug report, @siljesu! We plan to first do a release, then merge https://github.com/lf-lang/lingua-franca/pull/1482, which will unblock https://github.com/lf-lang/lingua-franca/pull/1534 which, in turn, is related to this issue. @patilatharva, would you be able to include a fix for this problem in #1534 (or a in a branch based off `picocli`)?

**Comment by patilatharva** [comment]
Yes! I'll try to include it in #1534 before we merge it in, but if not, in a separate branch based off `picocli`.

**Comment by lhstrh** [comment]
I realized the problem is actually not in the processing of CLI arguments. There's a fix in #1604 that is pending for review.
